### PR TITLE
[DOCS] Scan yaml examples for class names that should be part of the public API

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Scan yaml examples for class names that should be a part of the public API.
- Bump the threshold to 87 to include the new items.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by Kilo59** [comment]
@anthonyburdi we are starting to rely on these script files more and more.
Can we bring them under our `ruff` linter coverage?
The sooner we do it, the easier it will be.

File: scripts/public_api_report.py [hunk 1]
```diff
@@ -164,7 +170,11 @@ def _get_names_of_all_usages_in_file(self, file_contents: FileContents) -> Set[s
         import_names = self._get_non_private_gx_import_names(imports=gx_imports)
         logger.debug(f"import_names: {import_names}")
 
-        return function_names | import_names
+        pattern = r"class_name: (\w+)"
```
Comment (Kilo59) [inline scripts/public_api_report.py]: Should get a slight performance boost by compiling this regex.

https://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile

Opinion seems a bit split on how much this improves performance. But it should be at least somewhat faster

**Review by donaldheppner — APPROVED** [review]

**Review by Kilo59 — APPROVED** [review]

File: scripts/public_api_report.py (hunk 1 above)
Comment (anthonyburdi) [reply scripts/public_api_report.py]: Good call! Done :)

**Comment by anthonyburdi** [comment]
> @anthonyburdi we are starting to rely on these script files more and more. Can we bring them under our `ruff` linter coverage? The sooner we do it, the easier it will be.

Good call - I like this idea. I'm sure we will end up adding more scripts... though this feels like a utils directory I'm not sure where else to put it. Maybe these files should be moved into the ci directory. And also put under linter coverage. I was wondering why it wasn't being more strict. Do we remove `scripts` from ruff `extend-excludes`: https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L245 and fix the issues? Or is there something else we need to change as well? (E.g. add `scripts` to mypy files? https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L29)

**Comment by anthonyburdi** [comment]
> > @anthonyburdi we are starting to rely on these script files more and more. Can we bring them under our `ruff` linter coverage? The sooner we do it, the easier it will be.
> 
> Good call - I like this idea. I'm sure we will end up adding more scripts... though this feels like a utils directory I'm not sure where else to put it. Maybe these files should be moved into the ci directory. And also put under linter coverage. I was wondering why it wasn't being more strict. Do we remove `scripts` from ruff `extend-excludes`: https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L245 and fix the issues? Or is there something else we need to change as well? (E.g. add `scripts` to mypy files? https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L29)

Done! https://github.com/great-expectations/great_expectations/pull/7270

**Pull request merged**